src/SAML2/Builder/AssertionBuilder.php

```diff
@@ -86,9 +85,11 @@ public function setNotBefore(\DateInterval $interval = null)
 
         $this->assertion->setNotBefore($beforeTime->getTimestamp());
 
-        $confirmation = $this->assertion->getSubjectConfirmation()[0];
-        $confirmation->SubjectConfirmationData->NotOnOrAfter = $beforeTime->getTimestamp();
-        $this->assertion->setSubjectConfirmation([$confirmation]);
+        if ($interval !== null) {
+            $confirmation = $this->assertion->getSubjectConfirmation()[0];
+            $confirmation->SubjectConfirmationData->NotOnOrAfter = $beforeTime->getTimestamp();
```

Comment:
Should be `NotBefore` no ?

Reply:
If we set it as null in the config than should not.
I have fixed an error:

SubjectConfirmationData notBefore must be null for Bearer confirmation
Jul 19, 2018 4:01:21 PM SEVERE org.pac4j.saml.sso.impl.SAML2DefaultResponseValidator validateSamlSSOResponse
Current assertion validation failed, continue with the next one
org.pac4j.saml.exceptions.SAMLException: Subject confirmation validation failed

Reply:
Yeah true

Reply:
So what need to be fixed to keep the test green ?

Reply:
I will look

Reply:
I have to fix the same bug:
- `AssertionBuilder.setNotBefore` should set `NotBefore` subjectConfirmationData and not `NotOnOrAfter`
So it will be hard to merge :|

Reply:
No worries, I will merge this one first then we can make sure the other PR is fine